internal/cloud/realm/client.go

```diff
@@ -17,16 +18,31 @@ const (
 // Client is a Realm client
 type Client interface {
 	Authenticate(publicAPIKey, privateAPIKey string) (AuthResponse, error)
+	GetUserProfile() (UserProfile, error)
+	FindProjectAppByClientAppID(groupIDs []string, app string) ([]App, error)
 	Status() error
 }
 
 // NewClient creates a new Realm client
 func NewClient(baseURL string) Client {
-	return &client{baseURL}
+	return &client{baseURL: baseURL}
+}
+
+// NewAuthClient creates a new Realm client with a session used for Authorization
+func NewAuthClient(baseURL string, session *Session) Client {
+	// TODO: REALMC-7156 should we return an error here?
```

Comment:
Not sure if we want some kind of validation on a session here, or if we want to use the refresh token when creating the client to get a fresh token

Reply:
you can keep this as is (minus the pointer to `Session`).  error/validation logic will be put at the point of request execution time: if auth (or refresh) is required for an api call and we don't see any value for the particular access/refresh token we need, that's when we can return a "must authenticate" error.